code/clement/django/lab02/usersystem_app/views.py

```diff
@@ -0,0 +1,78 @@
+from django.shortcuts import render, redirect
+
+from django.contrib.auth import (
+    authenticate,
+    login as blog_login,
+    logout as blog_logout,
+    get_user_model,
+)
+from django.contrib.auth.decorators import login_required
+from django.contrib import messages
+
+from .models import Usersystem
+from django.http import HttpResponse
+
+
+# Create your views here.
+def register(request):
+    if request.method == 'GET':
+        return render(request, 'users/register.html')
+
+    elif request.method == 'POST':
+        form = request.POST
+        username = form.get('username')
+        password = form.get('password')
+        print(form)
+
+# use the form data to create a new user object (use the create_user method instead of the create method)
+        new_usersystem = Usersystem.objects.create_user(
+            username=username,
+            password=password,
+        )
+
+        # login the new user (using django's login function)
+        messages.success(request, f'Welcome, {new_usersystem.username}')
+
+        blog_login(request, new_usersystem)
+        return redirect('usersystem_app:index')
```

Comment:
With the way the urls are set up for the project/app, the register view should redirect to the profile view